driver-core/src/main/com/mongodb/operation/OperationHelper.java

```diff
@@ -274,6 +275,9 @@ static boolean canRetryWrite(final ServerDescription serverDescription, final Co
         } else if (connectionDescription.getServerType().equals(ServerType.STANDALONE)) {
             LOGGER.debug("retryWrites set to true but the server is a standalone server.");
             return false;
+        } else if (!sessionContext.hasSession()) {
```

Comment:
Adding this check here rather than before line 262 in `isRetryableWrite` because otherwise this check will shadow the conditions checked in this method and make the log message less useful